driver/util.c

```diff
@@ -513,19 +500,30 @@ WnbdProcessDeviceThreadReplies(_In_ PSCSI_DEVICE_INFORMATION DeviceInformation)
         CloseConnection(DeviceInformation);
         goto Exit;
     }
-    WNBD_LOG_LOUD("Received reply header for %p 0x%llx.", Element->Srb, Element->Tag);
 
     ULONG StorResult;
     if(!Element->Aborted) {
-        StorResult = StorPortGetSystemAddress(Element->DeviceExtension, Element->Srb, &SrbBuff);
-        if (STOR_STATUS_SUCCESS != StorResult) {
-            WNBD_LOG_ERROR("Could not get SRB %p 0x%llx data buffer. Error: %d.",
-                           Element->Srb, Element->Tag, error);
-            Element->Srb->SrbStatus = SRB_STATUS_INTERNAL_ERROR;
-            CloseConnection(DeviceInformation);
-            goto Exit;
+        // We need to avoid accessing aborted or already completed SRBs.
+        PCDB Cdb = (PCDB)&Element->Srb->Cdb;
+        int NbdReqType = ScsiOpToNbdReqType(Cdb->AsByte[0]);
+        WNBD_LOG_LOUD("Received reply header for %s %p 0x%llx.",
+                      NbdRequestTypeStr(NbdReqType), Element->Srb, Element->Tag);
+
+        if(IsReadSrb(Element->Srb)) {
+            StorResult = StorPortGetSystemAddress(Element->DeviceExtension, Element->Srb, &SrbBuff);
+            if (STOR_STATUS_SUCCESS != StorResult) {
+                WNBD_LOG_ERROR("Could not get SRB %p 0x%llx data buffer. Error: %d.",
+                               Element->Srb, Element->Tag, error);
+                Element->Srb->SrbStatus = SRB_STATUS_INTERNAL_ERROR;
+                CloseConnection(DeviceInformation);
+                goto Exit;
+            }
         }
     }
+    else {
```

Comment:
We don't have a proper coding style documented. Can you please use the form:
if () {
} else {
}
It will be consistent we what we have already.